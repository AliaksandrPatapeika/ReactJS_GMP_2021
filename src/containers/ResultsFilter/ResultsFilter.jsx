import './ResultsFilter.less';

import React, {useState} from 'react';

import FilterItem from '../../components/FilterItem';

const ResultsFilter = () => {
  const filterItems = [
    {id: 1, name: 'ALL'},
    {id: 2, name: 'DOCUMENTARY'},
    {id: 3, name: 'COMEDY'},
    {id: 4, name: 'HORROR'},
    {id: 5, name: 'CRIME'}
  ];
  const [activeFilterId, setActiveFilterId] = useState(1);

  const setFilterId = (id) => {
    setActiveFilterId(id);
  };

  return (
    <nav>
      <ul className="resultsFilterContainer">
        {filterItems.map((item) => (
          <FilterItem
            name={item.name}
            key={item.id}
            isActive={item.id === activeFilterId}
            setFilterId={() => setFilterId(item.id)}
          />
        ))}
      </ul>
    </nav>
  );
};

export default ResultsFilter;
