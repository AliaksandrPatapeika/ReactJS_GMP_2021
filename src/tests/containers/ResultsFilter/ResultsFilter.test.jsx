import React from 'react';
import renderer from 'react-test-renderer';

import ResultsFilter from '../../../containers/ResultsFilter';

describe('ResultsFilter', () => {
  test('renders correctly', () => {
    const resultsFilter = renderer
      .create(<ResultsFilter />)
      .toJSON();

    expect(resultsFilter).toMatchSnapshot();
  });
});
