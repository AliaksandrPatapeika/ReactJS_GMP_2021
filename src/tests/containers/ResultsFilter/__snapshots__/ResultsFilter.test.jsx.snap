// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsFilter renders correctly 1`] = `
<nav>
  <ul
    className="resultsFilterContainer"
  >
    <li
      className="filterItem active"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
    >
      ALL
    </li>
    <li
      className="filterItem"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
    >
      DOCUMENTARY
    </li>
    <li
      className="filterItem"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
    >
      COMEDY
    </li>
    <li
      className="filterItem"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
    >
      HORROR
    </li>
    <li
      className="filterItem"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="menuitem"
    >
      CRIME
    </li>
  </ul>
</nav>
`;
